packages/pluggableWidgets/datagrid-web/src/components/ColumnSelector.tsx

```diff
@@ -19,11 +20,40 @@ export function ColumnSelector(props: ColumnSelectorProps): ReactElement {
 
     useOnClickOutside([buttonRef, optionsRef], () => setShow(false));
 
+    const onClick = useCallback(
+        (isVisible: boolean, id: string) =>
+            props.setHiddenColumns(prev => {
+                if (!isVisible) {
+                    prev.splice(
+                        prev.findIndex(v => v === id),
+                        1
+                    );
```

Comment:
Can't this just be `return prev.filter(v => v !== id)`? That also immediately creates a new array. (I know u copied this, but just wondering now as well)

Reply:
Actually the other filters (dropdown) copied this logic.. so this is the original starting point 😢